### PR TITLE
Add NOAA C-CAP colormap

### PR DESCRIPTION
**Pull request by gadomski**
## Description

Add colormap for NOAA C-CAP dataset. [Here's the stactools package](https://github.com/stactools-packages/noaa-c-cap).

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

It hasn't. No tests for colormaps AFAICT.

## Checklist:

Please delete options that are not relevant.

- [x] I have performed a self-review
- [x] Code is linted and styled ~(./scripts/format)~ (`black pctiler/pctiler`)

### CODE REVIEW

File: pctiler/pctiler/colormaps/__init__.py [hunk 1]
```diff
@@ -22,6 +23,7 @@
     **lulc_colormaps,
     **mtbs_colormaps,
     **chloris_colormaps,
+    "noaa-c-cap": noaa_c_map_colormap,
```
Comment (lossyrob) [inline pctiler/pctiler/colormaps/__init__.py:26]: @mmcfarland do you have a preference around maintaining the `**x_colormaps` approach for consistency?

Comment (mmcfarland) [reply pctiler/pctiler/colormaps/__init__.py:26]: I would prefer that, especially if we may over time be adding noaa related cmaps

File: pctiler/pctiler/colormaps/noaa_c_cap.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    16: (57, 0, 57, 255),
+    17: (108, 0, 108, 255),
+    18: (181, 0, 181, 255),
+    19: (0, 235, 235, 255),
+    20: (235, 235, 0, 255),
+    21: (0, 0, 120, 255),
+    22: (0, 0, 236, 255),
+    23: (102, 117, 219, 255),
+    24: (1, 26, 119, 255),
+    25: (24, 24, 24, 255),
```
Comment (mmcfarland) [inline pctiler/pctiler/colormaps/noaa_c_cap.py:30]: I'm curious if these values beyond `25` are necessary for the data set? Looks like this is a land cover change raster and the first 24 are categories? We don't need to scale to 255 if those values aren't present in the data, and it'll make dynamic legend generation easier if they are not.

with:
image

without:
image

Comment (mmcfarland) [reply pctiler/pctiler/colormaps/noaa_c_cap.py:30]: Related to above: if this is categorical, just confirming that the STAC collection will include the value/labels in the `item_assets.file:values` extension, similar to [io-lulc](https://planetarycomputer-staging.microsoft.com/api/stac/v1/collections/io-lulc)

Comment (gadomski) [reply pctiler/pctiler/colormaps/noaa_c_cap.py:30]: > I would prefer that, especially if we may over time be adding noaa related cmaps

Roger, will fix.

> I'm curious if these values beyond 25 are necessary for the data set? 

Shouldn't be, I just copied from the TIFF's internal map. I'll double-check manually and edit as appropriate.

>  Related to above: if this is categorical, just confirming that the STAC collection will include the value/labels in the item_assets.file:values extension, similar to io-lulc

Hm, I forgot about that (I just did `label:classes` for categories). I'll fix on the collection side, thanks for calling that out.

**Comment by gadomski** [comment]
@mmcfarland @lossyrob ready for re-review.

**Review by mmcfarland — APPROVED** [review]
Perfect, thanks @gadomski!

**Pull request merged**